tsdb/head.go

```diff
@@ -629,6 +688,55 @@ func (h *Head) Init(minValidTime int64) error {
 	return nil
 }
 
+func (h *Head) loadMmappedChunks() (map[uint64][]*mmappedChunk, error) {
+	mmappedChunks := map[uint64][]*mmappedChunk{}
+	if err := h.chunkDiskMapper.IterateAllChunks(func(seriesRef, chunkRef uint64, mint, maxt int64, numSamples uint16) error {
+		slice := mmappedChunks[seriesRef]
+		if len(slice) > 0 {
+			if slice[len(slice)-1].maxTime >= mint {
+				return errors.Errorf("out of sequence m-mapped chunk for series ref %d", seriesRef)
+			}
+		}
+
+		slice = append(slice, &mmappedChunk{
+			ref:        chunkRef,
+			minTime:    mint,
+			maxTime:    maxt,
+			numSamples: numSamples,
+		})
+		mmappedChunks[seriesRef] = slice
+
+		h.metrics.chunks.Inc()
+		h.metrics.chunksCreated.Inc()
+		return nil
+	}); err != nil {
+		return nil, errors.Wrap(err, "iterate on-disk chunks")
+	}
+	return mmappedChunks, nil
+}
+
+// repairMmappedChunks attempts repair of the mmapped chunks and if it fails, it clears all the previously
+// loaded mmapped chunks.
+func (h *Head) repairMmappedChunks(err error) map[uint64][]*mmappedChunk {
```

Comment:
If we want to be consistent, then `wal.Repair` should also be renamed, because WAL repair basically deletes the corrupted data. Sounds good? 

Also, I kept is separate to keep away the inner details from Init and make Init more readable on a higher level, so I would prefer keeping it separate.

Reply:
I renamed only for mmapped chunks now, WAL can be another PR, lots of places to change for that